File: app/src/main/java/com/example/nasmovie/ui/adapter/SearchResultAdapter.java
package com.example.nasmovie.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.nasmovie.R;
import com.example.nasmovie.data.model.Movie;
import com.example.nasmovie.view.MovieCard;

import java.util.ArrayList;
import java.util.List;

/**
 * 搜索结果适配器 - 3列网格布局
 */
public class SearchResultAdapter extends RecyclerView.Adapter<SearchResultAdapter.ViewHolder> {

    private List<Movie> movies = new ArrayList<>();
    private OnMovieClickListener listener;

    public interface OnMovieClickListener {
        void onMovieClick(Movie movie);
    }

    public void setOnMovieClickListener(OnMovieClickListener listener) {
        this.listener = listener;
    }

    public void setMovies(List<Movie> movies) {
        this.movies = movies != null ? movies : new ArrayList<>();
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_search_result, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Movie movie = movies.get(position);
        holder.bind(movie);
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        private final MovieCard movieCard;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            movieCard = itemView.findViewById(R.id.movie_card);

            itemView.setOnClickListener(v -> {
                int position = getBindingAdapterPosition();
                if (position != RecyclerView.NO_POSITION && listener != null) {
                    listener.onMovieClick(movies.get(position));
                }
            });
        }

        void bind(Movie movie) {
            movieCard.setMode(MovieCard.CardMode.SEARCH);
            movieCard.loadMovie(movie);
        }
    }
}

File: app/src/main/java/com/example/nasmovie/ui/adapter/SearchResultAdapter.kt
package com.example.nasmovie.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.example.nasmovie.data.model.Movie
import com.example.nasmovie.databinding.ItemSearchResultBinding
import com.example.nasmovie.view.MovieCard

/**
 * 搜索结果适配器 - 3列网格布局 (Kotlin + ListAdapter)
 */
class SearchResultAdapter : ListAdapter<Movie, SearchResultAdapter.ViewHolder>(MovieDiffCallback()) {

    private var listener: OnMovieClickListener? = null

    interface OnMovieClickListener {
        fun onMovieClick(movie: Movie)
    }

    fun setOnMovieClickListener(listener: OnMovieClickListener) {
        this.listener = listener
    }

    // 兼容旧代码
    fun setMovies(movies: List<Movie>?) {
        submitList(movies ?: emptyList())
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemSearchResultBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class ViewHolder(private val binding: ItemSearchResultBinding) : RecyclerView.ViewHolder(binding.root) {

        init {
            binding.root.setOnClickListener {
                val position = bindingAdapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    listener?.onMovieClick(getItem(position))
                }
            }
        }

        fun bind(movie: Movie) {
            binding.movieCard.setMode(MovieCard.CardMode.SEARCH)
            binding.movieCard.loadMovie(movie)
        }
    }

    private class MovieDiffCallback : DiffUtil.ItemCallback<Movie>() {
        override fun areItemsTheSame(oldItem: Movie, newItem: Movie): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: Movie, newItem: Movie): Boolean {
            return oldItem == newItem
        }
    }
}
